bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpTargetHandler.java

```diff
@@ -0,0 +1,376 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.snmp.internal;
+
+import static org.openhab.binding.snmp.internal.SnmpBindingConstants.*;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.snmp.internal.config.SnmpChannelConfiguration;
+import org.openhab.binding.snmp.internal.config.SnmpTargetConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.snmp4j.AbstractTarget;
+import org.snmp4j.CommandResponder;
+import org.snmp4j.CommandResponderEvent;
+import org.snmp4j.CommunityTarget;
+import org.snmp4j.PDU;
+import org.snmp4j.event.ResponseEvent;
+import org.snmp4j.event.ResponseListener;
+import org.snmp4j.mp.SnmpConstants;
+import org.snmp4j.smi.Counter64;
+import org.snmp4j.smi.Integer32;
+import org.snmp4j.smi.IpAddress;
+import org.snmp4j.smi.OID;
+import org.snmp4j.smi.OctetString;
+import org.snmp4j.smi.UdpAddress;
+import org.snmp4j.smi.UnsignedInteger32;
+import org.snmp4j.smi.Variable;
+import org.snmp4j.smi.VariableBinding;
+
+/**
+ * The {@link SnmpTargetHandler} is responsible for handling commands, which are
+ * sent to one of the channels or update remote channels
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class SnmpTargetHandler extends BaseThingHandler implements ResponseListener, CommandResponder {
+    private final Logger logger = LoggerFactory.getLogger(SnmpTargetHandler.class);
+
+    private @NonNullByDefault({}) SnmpTargetConfiguration config;
+    private final SnmpService snmpService;
+    private @NonNullByDefault({}) ScheduledFuture<?> refresh;
```

Comment:
AFAIK these are not candidates for this annotation or did I miss some discussions?

Reply:
https://www.openhab.org/docs/developer/guidelines.html#null-annotations

Reply:
My understanding was like https://github.com/openhab/openhab-core/pull/910#discussion_r300031597 this: every member that we know is not null before using and this can't be detected by the tooling. This is the case here. But I can add null-checks and make it `@Nullable`if you want.

Reply:
But IMHO @maggu2810 has the comment as I have :-)

Reply:
> But I can add null-checks and make it @Nullableif you want.

If its null at some point then, yes please